### PR TITLE
[Drupal] Patches for content editors

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR does a couple tweaks to how our Drupal setup.

- Add the `/drupal` prefix into localhost builds, so that Drupal pages don't override pages in Heroku review instances.
- Add a `draft` frontmatter property, so new pages can be merged to master without being published. This way, during its next content scrape, Drupal can pick up that page as a draft.

Ticket - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17477
## Testing done
- Local testing=

## Acceptance criteria
- [ ] Editors can see their healthcare edits in Heroku
- [ ] Editors can merge new pages into `master` without worry about it being published

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/plugins/create-environment-filter.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
   const environmentName = options.buildtype;
 
   return (files, metalsmith, done) => {
     for (const fileName of Object.keys(files)) {
       const file = files[fileName];
+
+      if (environmentName !== ENVIRONMENTS.LOCALHOST) {
+        if (file.draft) delete files[fileName];
+      }
+
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-environment-filter.js]: There's more context in the ticket, but this `draft` property makes it safe for content editors to merge unfinished pages into `master` for Drupal to pick up.

File: src/site/constants/drupals.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ const ENABLED_ENVIRONMENTS = new Set([
 ]);
 
 const PREFIXED_ENVIRONMENTS = new Set([
-  // ENVIRONMENTS.LOCALHOST,
+  ENVIRONMENTS.LOCALHOST,
```
Comment (ncksllvn) [inline src/site/constants/drupals.js]: Without the prefix on localhost, content team members can't preview their changes in Heroku on vagov-content.

**Review by beeyayjay — APPROVED** [review]
LGTM

**Review by jbalboni — APPROVED** [review]

**Comment by ncksllvn** [comment]
Going to merge this now and it's conflicting with content work

**Pull request merged**